### PR TITLE
Support ld+json website feeds starting with OutSavvy

### LINKED ISSUE #1367: Support OutSavvy by implementing ld+json support
## User story

As a partner using OutSavvy, I want my events to be imported into PlaceCal automatically, so I can get them listed without any extra effort

## Acceptance criteria

- [ ] Outsavvy event listings can be imported
- [x] ~README and example environement shows how to set it up - no longer needed~
- [ ] Draft handbook page is created

Current partners to support:

```
https://www.outsavvy.com/organiser/sappho-events
https://www.outsavvy.com/organiser/lez-events
```

## Implementation notes & questions

[They have an API](https://www.outsavvy.com/organiser/sappho-events) that requires an API key but implies it's global access once you're in. So at least we don't need per-user API keys for this.

I think it'll be easier to just use ld+json to scrape the pages we want though

## Implementation plan

- [ ] Figure out how to scrape ld+json from the page
- [ ] Make a new importer archetype for ld+json
- [ ] Add specific tweaks for this

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Fixes #1367

### CODE REVIEW

File: app/controllers/partners_controller.rb [hunk 1]
```diff
@@ -21,8 +21,26 @@ def index
     # Get all partners based in the neighbourhoods associated with this site.
     @partners = Partner.for_site(current_site).order(:name)
 
-    # show only partners with no service_areas
-    @map = get_map_markers(@partners, true) if @partners.detect(&:address)
+    respond_to do |format|
```
Comment (kimadactyl) [inline app/controllers/partners_controller.rb:24]: is this meant to be part of this PR?

File: app/jobs/calendar_importer/parsers/outsavvy.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+# In order for a parser to be recognized, it must be added
+# to the PARSERS constant list in app/models/calendar_parser.rb.
+# Parent parser classes should not be added.
+
+module CalendarImporter::Parsers
+  class OutSavvy < Base
+    NAME = 'OutSavvy'
+    KEY = 'outsavvy'
+    DOMAINS = %w[www.outsavvy.com].freeze
```
Comment (kimadactyl) [inline app/jobs/calendar_importer/parsers/outsavvy.rb:11]: is it always www?

File: config/database.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ default: &default
 
 development:
   <<: *default
-  database: placecal_dev
+  database: placecal_dev_prod_db
```
Comment (kimadactyl) [inline config/database.yml:14]: is this meant to be here?

File: .prettierignore [hunk 4]
```diff
@@ -5,3 +5,4 @@
 /tmp/
 /vendor/bundle/
 /.rubocop_todo.yml
+outsavvy-sapho-events.html
```
Comment (kimadactyl) [inline .prettierignore:8]: not sure this file should be here?

File: app/jobs/calendar_importer/parsers/out_savvy.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+# In order for a parser to be recognized, it must be added
+# to the PARSERS constant list in app/models/calendar_parser.rb.
+# Parent parser classes should not be added.
+
+module CalendarImporter::Parsers
+  class OutSavvy < Base
+    NAME = 'OutSavvy'
+    KEY = 'out-savvy'
+    DOMAINS = %w[www.outsavvy.com].freeze
```
Comment (kimadactyl) [inline app/jobs/calendar_importer/parsers/out_savvy.rb:11]: do we need this www?

**Comment by kimadactyl** [comment]
- [ ] There is a `outsavvy-sapho-events.html` in the root I don't think should be there but can't leave a comment on it

File: .prettierignore (hunk 4 above)
Comment (ivan-kocienski-gfsc) [reply .prettierignore:8]: This was a draft PR holdover when F wanted to see how far i had gotten

File: app/jobs/calendar_importer/parsers/out_savvy.rb (hunk 5 above)
Comment (ivan-kocienski-gfsc) [reply app/jobs/calendar_importer/parsers/out_savvy.rb:11]: This is just the thing that is shown to the user on the calendar page

**Review by kimadactyl — APPROVED** [review]
Approving so you can merge when happy

**Pull request merged**